src/prefect/cli/deploy.py

```diff
@@ -721,10 +722,156 @@ def _merge_with_default_deploy_config(deploy_config: Dict):
+                repo_token = prompt(
+                    prompt_message,
+                    console=console,
+                )
+            await Secret(
+                value=repo_token,
+            ).save(name=token_secret_block_name, overwrite=True)
+
+    git_clone_step = {
+        "prefect.deployments.steps.git_clone": {
+            "repository": remote_url,
+            "branch": branch,
+        }
+    }
+
+    if token_secret_block_name:
+        git_clone_step["prefect.deployments.steps.git_clone"]["access_token"] = (
+            "{{ prefect.blocks.secret." + token_secret_block_name + " }}"
+        )
+
+    return [git_clone_step]
+
+
+async def _generate_pull_step_for_build_docker_image(
+    console: Console, deploy_config: Dict, auto: bool = True
+):
+    pull_step = {}
+    if auto:
+        pull_step["path"] = "/opt/prefect/{{ name }}"
+    else:
+        pull_step["path"] = prompt(
+            "What is the path to your flow code in your Dockerfile?",
+            default="/opt/prefect/{{ name }}",
+            console=console,
+        )
+
+    return [{"prefect.deployments.steps.set_working_directory": pull_step}]
+
+
+async def _check_for_build_docker_image_step(
+    build_action: List[Dict],
+) -> Optional[Dict[str, Any]]:
+    if not build_action:
+        return None
+
+    build_docker_image_steps = [
+        "prefect_docker.projects.steps.build_docker_image",  # legacy
+        "prefect_docker.deployments.steps.build_docker_image",
+    ]
+    for build_docker_image_step in build_docker_image_steps:
+        for action in build_action:
+            if action.get(build_docker_image_step):
+                return action.get(build_docker_image_step)
+
+    return None
+
+
 async def _generate_default_pull_action(
-    console: Console, deploy_config: Dict, ci: bool = False
+    console: Console, deploy_config: Dict, actions: List[Dict], ci: bool = False
 ):
     remote_url = _get_git_remote_origin_url()
+    build_docker_image_step = await _check_for_build_docker_image_step(actions["build"])
+
+    if build_docker_image_step:
+        dockerfile = build_docker_image_step.get("dockerfile")
+        if dockerfile == "auto":
+            return await _generate_pull_step_for_build_docker_image(
+                console, deploy_config
+            )
+        else:
+            if is_interactive():
+                if remote_url and confirm(
+                    "Would you like to pull your flow code from its remote"
+                    " repository when running your deployment?"
+                ):
+                    return await _generate_git_clone_pull_step(
+                        console, deploy_config, remote_url
+                    )
+                if not confirm("Do you copy your flow code in your Dockerfile?"):
+                    exit_with_error(
+                        "Your flow code must be copied into your Docker image"
+                        " in order to run your deployment."
```

Comment:
Suggestion:

Your flow code must be copied into your Docker image to run your deployment.

To copy your flow code into your Docker image you can this line to your Dockerfile:

    COPY . /opt/prefect/

Reply:
#nodeadends